### DISCUSSION TITLE
Event trigger based on attribute originEntity vs currentNote vs startNote

### DISCUSSION THREAD

**Discussion by daranable** [TriliumNext/Trilium#1856] [Q&A] [answered]
Hi! Once again trying to figure something out,

I've gotten the runOnAttributeChange event firing, however I was expecting one of the above api values to show me the note that had the attribute changed in it.  However when the event fires and I log currentNote and startNote to the backend, the both are just the script that is currently running "onDinnerChanged".

For context I am wanting to retrieve the note that had the attribute changed so I can copy the value and put it in a JSON file, I already am able to update the JSON data note without issue, but haven't managed to get the correct note to pull data from.

Thanks again for the help.

**Comment by zadam** [comment, marked as answer]
For this purpose there's `originEntity` which contains the entity which sort of triggered the event. In the case of runOnAttributeChange the originEntity is the attribute on which the change is being done. If you want to access the note owning this attribute then do just:

```javascript
const note = api.originEntity.getNote();
```

**Reply by daranable** [reply]
Thanks! originEntity was definitely what I was looking for, when I tested originEntity, I hadn't realized it was of the Attribute type, that helps a lot.

Best Regards,
-Daranable

**Reply by zadam** [reply]
Yeah, type of originEntity depends on the event, that's why its name is so generic ...
